src/pt/spectralscan/src/eu/kanade/tachiyomi/extension/pt/spectralscan/NexusScan.kt

```diff
@@ -0,0 +1,311 @@
+
+                if (page.has_next) {
+                    fetchRemainingChapters(slug, allChapters, currentPage + 1)
+                } else {
+                    Observable.just(allChapters)
+                }
+            }
+    }
+
+    private fun parseChaptersFromHtml(html: String): List<SChapter> {
+        val document = Jsoup.parse(html, baseUrl)
+        return document.select("a.chapter-item[href]").map { element ->
+            SChapter.create().apply {
+                name = element.selectFirst(".chapter-number")?.text()?.trim()
+                    ?: element.attr("data-chapter-number")?.let { "Capítulo $it" }
+                    ?: "Capítulo"
+                setUrlWithoutDomain(element.absUrl("href").ifEmpty { element.attr("href") })
+
+                element.selectFirst(".chapter-date")?.text()?.let { dateStr ->
+                    date_upload = parseChapterDate(dateStr)
+                }
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val slug = getMangaSlug(manga.url)
+        val url = "$baseUrl/ajax/load-chapters/".toHttpUrl().newBuilder()
+            .addQueryParameter("item_slug", slug)
+            .addQueryParameter("page", "1")
+            .addQueryParameter("sort", "desc")
+            .addQueryParameter("q", "")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response) =
+        parseChaptersFromHtml(response.parseAs<ChapterListResponse>().chapters_html)
+
+    private fun parseChapterDate(dateStr: String): Long {
+        return try {
+            when {
+                dateStr.contains("hoje", ignoreCase = true) -> System.currentTimeMillis()
+                dateStr.contains("ontem", ignoreCase = true) -> System.currentTimeMillis() - 24 * 60 * 60 * 1000
+                dateStr.contains("hora", ignoreCase = true) -> {
+                    val hours = Regex("(\\d+)").find(dateStr)?.value?.toIntOrNull() ?: 1
+                    System.currentTimeMillis() - hours * 60 * 60 * 1000
+                }
+                dateStr.contains("dia", ignoreCase = true) -> {
+                    val days = Regex("(\\d+)").find(dateStr)?.value?.toIntOrNull() ?: 1
+                    System.currentTimeMillis() - days * 24 * 60 * 60 * 1000
+                }
+                dateStr.contains("semana", ignoreCase = true) -> {
+                    val weeks = Regex("(\\d+)").find(dateStr)?.value?.toIntOrNull() ?: 1
+                    System.currentTimeMillis() - weeks * 7 * 24 * 60 * 60 * 1000
+                }
+                dateStr.contains("mês", ignoreCase = true) || dateStr.contains("mes", ignoreCase = true) -> {
+                    val months = Regex("(\\d+)").find(dateStr)?.value?.toIntOrNull() ?: 1
+                    System.currentTimeMillis() - months * 30L * 24 * 60 * 60 * 1000
+                }
```

Comment:
Consider using `java.util.Calendar`, you can see an example in Love4u. This is not the only example to draw inspiration from, just the first one I found. Could also check Zbulu, MangaGun, KomikCast, etc.
https://github.com/keiyoushi/extensions-source/blob/c00f88ee7f3e3c762af33c15e567a68b6aa877ce/src/ja/welovemangaone/src/eu/kanade/tachiyomi/extension/ja/welovemangaone/Love4u.kt#L43-L61